resources/osl_mysql_dev.rb

```diff
@@ -0,0 +1,31 @@
+provides :osl_mysql_dev
+resource_name :osl_mysql_dev
+unified_mode true
+
+default_action :create
+
+property :database, String, name_property: true
+property :username, String, required: true
+property :password, String, required: true
+
+# Install the mariadb package, set up the service, set up the user, then set up the given database
+action :create do
+  # Install the package, and set up the service
+  mariadb_server_install 'MariaDB' do
+    password 'iloveinsecurepasswords'
+    action [:install, :create]
+  end
+  # Create new database
+  mariadb_database new_resource.database do
+    password 'iloveinsecurepasswords'
+    encoding 'utf8mb4'
+    collation 'utf8mb4_unicode_ci'
```

Comment:
Are both of these required? Perhaps it might be better to create a new parameter called `database_parameters` which is a hash of parameters that could be passed to this resource. You can just loop through the hash like the following:
```ruby
new_resource.database_parameters.each do |key, value|
  send(key.to_sym, value)
end
```

Reply:
My reasoning behind that specific encoding/collation is to make sure that the database was able to hold any character needed, as the standard utf-8 is still lacking Chinese, emojis, etc.

I'll be adding in the ability to dynamically add in more config options to the database instance.